src/components/sidebar/Sidebar.tsx

```diff
@@ -95,32 +97,41 @@ const pagesProperties: PageProperty[] = [
   },
 ];
 
-//-- Mocked fake data just for testing purpose --//
-const isVoyageStarted: boolean = true;
-//-- --//
-
 export default function Sidebar() {
   const currentPath = usePathname();
 
   const [isOpenSidebar, setIsOpenSidebar] = useState<boolean>(true);
   const [selectedButton, setSelectedButton] = useState<string>(currentPath);
   const [hoveredButton, setHoveredButton] = useState<string | null>(null);
 
-  const handlePageClick = (element: PageProperty | string) => {
-    if (typeof element === "string") {
-      setSelectedButton(element);
-    } else if (element.name === MainPages.myVoyage) {
-      setIsOpenSidebar(true);
-    } else {
-      setSelectedButton(element.link);
-    }
-  };
+  const { isAuthenticated } = useAppSelector(
+    (state: RootState): { isAuthenticated: boolean } => state.auth,
+  );
+
+  const isVoyageStarted: boolean = isAuthenticated;
+
+  useEffect(() => {
+    setSelectedButton(currentPath);
+  }, [currentPath]);
+
+  const handlePageClick = useCallback(
```

Comment:
I think it'd be okay to remove `setSelectedButton` because `useEffect` already updates `selectedButton` when current path changes (a user clicks a link). Only line 121 and 122 are needed. @timothyrusso  What do you think?

Reply:
Also one small thing I noticed there is `const currentPath = usePathname();` in `PageButton.txt`. Maybe I'm missing smth but I don't think you need to call it again because `selectedButton` is a current path 🤓

Reply:
Thanks @JaneMoroz , I've removed the currentPath from the PageButton.
For the handlePageClick I'm having some typescript problems if I modify this logic, I'll work on it an try to solve the problem.